sbi/inference/snpe/snpe_base.py

```diff
@@ -82,7 +85,10 @@ def __init__(
         self._summary.update({"rejection_sampling_acceptance_rates": []})  # type:ignore
 
     def append_simulations(
-        self, theta: Tensor, x: Tensor, proposal: Optional[Any] = None
+        self,
+        theta: Tensor,
+        x: Tensor,
+        proposal: Optional[DirectPosterior] = None,
```

Comment:
Couldn't the proposal be sth else than a `DirectPosterior`, e.g., a `nn.Module`?

Reply:
It has to be a DirectPosterior because it must have a default_x